service/dynamodb/expression/expression.go

```diff
@@ -0,0 +1,600 @@
+//
+// Example:
+//
+//     // let expression be an instance of Expression{}
+//
+//     deleteInput := dynamodb.DeleteItemInput{
+//       ConditionExpression:       expression.Condition(),
+//       ExpressionAttributeNames:  expression.Names(),
+//       ExpressionAttributeValues: expression.Values(),
+//       Key: map[string]*dynamodb.AttributeValue{
+//         "PartitionKey": &dynamodb.AttributeValue{
+//           S: aws.String("SomeKey"),
+//         },
+//       },
+//       TableName: aws.String("SomeTable"),
+//     }
+func (e Expression) Condition() *string {
+	return e.returnExpression(condition)
+}
+
+// Filter will return the *string corresponding to the Filter Expression of the
+// argument Expression. This method is used to satisfy the members of DynamoDB
+// input structs. If the argument Expression does not have a FilterExpression,
+// Filter() will return nil.
+//
+// Example:
+//
+//     // let expression be an instance of Expression{}
+//
+//     queryInput := dynamodb.QueryInput{
+//       KeyConditionExpression:    expression.KeyCondition(),
+//       FilterExpression:          expression.Filter(),
+//       ExpressionAttributeNames:  expression.Names(),
+//       ExpressionAttributeValues: expression.Values(),
+//       TableName: aws.String("SomeTable"),
+//     }
+func (e Expression) Filter() *string {
+	return e.returnExpression(filter)
+}
+
+// Projection will return the *string corresponding to the Projection Expression
+// of the argument Expression. This method is used to satisfy the members of
+// DynamoDB input structs. If the argument Expression does not have a
+// ProjectionExpression, Projection() will return nil.
+//
+// Example:
+//
+//     // let expression be an instance of Expression{}
+//
+//     queryInput := dynamodb.QueryInput{
+//       KeyConditionExpression:    expression.KeyCondition(),
+//       ProjectionExpression:      expression.Projection(),
+//       ExpressionAttributeNames:  expression.Names(),
+//       ExpressionAttributeValues: expression.Values(),
+//       TableName: aws.String("SomeTable"),
+//     }
+func (e Expression) Projection() *string {
+	return e.returnExpression(projection)
+}
+
+// Names will return the map[string]*string corresponding to the
+// ExpressionAttributeNames of the argument Expression. This method is used to
+// satisfy the members of DynamoDB input structs. If Names() is called on an
+// empty Expression struct, it will return an empty map
+//
+// Example:
+//
+//     // let expression be an instance of Expression{}
+//
+//     queryInput := dynamodb.QueryInput{
+//       KeyConditionExpression:    expression.KeyCondition(),
+//       ProjectionExpression:      expression.Projection(),
+//       ExpressionAttributeNames:  expression.Names(),
+//       ExpressionAttributeValues: expression.Values(),
+//       TableName: aws.String("SomeTable"),
+//     }
+func (e Expression) Names() map[string]*string {
+	aliasList, _ := e.buildChildTrees()
```

Comment:
This should be done inside of `Builder.Build` so that any errors can be surfaced to the user. In addition, this will reduce the duplicate logic that is occurring in each getter. I think it makes sense to see the `Expression` type as a read only view into the already built/constructed data created by `Builder`